application/services/Shipments.php

```diff
@@ -769,7 +774,11 @@ public function updateTbResults($params) {
                 if ($mtbDetected != "error") {
                     $errorCode = null;
                     if(!in_array($mtbDetected, array("detected", "high", "medium", "low", "veryLow")) && ($rifResistance == null || $rifResistance == "")) {
-                        $rifResistance = "na";
+                        if($mtbDetected=="trace") {
+                            $rifResistance = "indeterminate";
+                        }else{
+                            $rifResistance = "na";
+                        }
```

Comment:
This logic looks good but there are multiple places in the application where is needs to be applied.